aws-sns-topic/src/main/java/software/amazon/sns/topic/BaseHandlerStd.java

```diff
@@ -202,7 +204,15 @@ private void invokeTagResource(final ProxyClient<SnsClient> proxyClient, final S
       // fail silently in case the user does not have access to either stack level or resource level tags
       logger.log(String.format("AccessDenied error: %s for topic: %s", e.getMessage(), topicArn));
     } catch (SnsException e) {
-      throw new CfnGeneralServiceException(e.getMessage(), e);
+      throw translateServiceExceptionToFailure(e);
     }
   }
+
+  private BaseHandlerException translateServiceExceptionToFailure(final SnsException ex) {
+    if (ex instanceof ThrottledException) {
+      return new CfnThrottlingException(ex); // CFN can retry on throttling error.
+    }
+    return new CfnGeneralServiceException(ex);
```

Comment:
Should we keep this as `new CfnGeneralServiceException(ex.getMessage(), ex)`?

Reply:
If you check the source code of that constructor in `CfnGeneralServiceException`, the first parameter is actually `operation`